src/lib/agent/LedgerService.ts

```diff
@@ -0,0 +1,181 @@
+import logger from '../logger';
+import { Wallet } from '../wallet/Wallet';
+
+export class LedgerService {
+  wallet: Wallet;
+  indy: Indy;
+  poolHandle?: PoolHandle;
+  authorAgreement?: AuthorAgreement;
+
+  constructor(wallet: Wallet, indy: Indy) {
+    this.wallet = wallet;
+    this.indy = indy;
+  }
+
+  async connect(poolName: string, poolConfig: PoolConfig) {
+    try {
+      logger.log(__dirname);
+
+      logger.log('Creating pool config');
+      await this.indy.createPoolLedgerConfig(poolName, poolConfig);
+    } catch (e) {
+      logger.log('PoolLedgerConfigAlreadyExists');
+      if (e.message !== 'PoolLedgerConfigAlreadyExistsError') {
+        throw e;
+      }
+    }
+
+    logger.log('Setting protocol version');
+    await this.indy.setProtocolVersion(2);
+
+    logger.log('Opening pool');
+    this.poolHandle = await this.indy.openPoolLedger(poolName);
+  }
+
+  async getPublicDid(myDid: Did) {
```

Comment:
As this is inside the ledger service I think we should just call it `getDid`. Every did we retrieve from the ledger is public.

Reply:
We should at least change `myDid` to `did` as I can retrieve DIDs that are not mine.

Reply:
I would still go with `getPublicDid`, there is a lot of Dids around when you're working with Indy/Aries. I agree with renaming `myDid` to `did` 👍

Reply:
I'm still in favour of naming it `getDid` as we already have `getPublicDid` on the agent, which returns our public did. 

For example this part from the tests where we call `getPublicDid` two times. One time on agent itself, one time in ledger service. Both do different things. I find it a bit confusing.

```
  test('get public DID from ledger', async () => {
    const { did } = faberAgent.getPublicDid();

    if (!did) {
      throw new Error('Agent does not have publid did.');
    }

    const result = await faberAgent.ledger.getPublicDid(did);
    expect(result).toEqual({
      did: faberAgentPublicDid.did,
      verkey: faberAgentPublicDid.verkey,
      role: '101',
    });
  });
```

But having mentioned my preference, I rest my case. If this is what we're discussing about we're definitely on the right track ;)